pkg/theme/theme.go

```diff
@@ -59,8 +61,34 @@ func UpdateTheme(themeConfig config.ThemeConfig) {
 	}
 }
 
+// singleHexColorToDec converts a hex color part to an int
+func singleHexColorToDec(hex string) int32 {
+	if len(hex) == 1 {
+		hex = hex + hex
+	}
+	i, err := strconv.ParseInt(hex, 16, 32)
+	if err != nil {
+		return -1
+	}
+	return int32(i)
+}
+
 // GetAttribute gets the gocui color attribute from the string
 func GetGocuiAttribute(key string) gocui.Attribute {
+	if len(key) > 0 && key[0] == '#' && (len(key) == 7 || len(key) == 4) {
```

Comment:
could we extract out this function out so that it takes a hex colour and returns the R,G and B components? Then we can add a couple of tests to verify it acts as expected for typical hex values

Reply:
Dune